apps/daimo-mobile/src/view/screen/deposit/BitrefillWebview.tsx

```diff
@@ -0,0 +1,111 @@
+import { assert, zDollarStr } from "@daimo/common";
+import { useContext } from "react";
+import { View } from "react-native";
+import { WebView } from "react-native-webview";
+import { getAddress, isAddress } from "viem";
+import { z } from "zod";
+
+import { DispatcherContext } from "../../../action/dispatch";
+import { useNav } from "../../../common/nav";
+import { i18NLocale } from "../../../i18n";
+import { ScreenHeader } from "../../shared/ScreenHeader";
+import { ss } from "../../shared/style";
+
+export function BitrefillWebView() {
+  const nav = useNav();
+  const dispatcher = useContext(DispatcherContext);
+
+  const handleMessage = (e: any) => {
+    const { event, paymentUri } = JSON.parse(e.nativeEvent.data);
+
+    switch (event) {
+      case "payment_intent": {
+        console.log(`[BITREFILL] payment_intent ${paymentUri}`);
+
+        const PaymentUriSchema = z.object({
+          protocol: z.literal("ethereum"),
+          address: z.string().refine((addr) => isAddress(addr), {
+            message: "Invalid Ethereum address",
+          }),
+          amount: zDollarStr,
+        });
+
+        try {
+          const [protocol, rest] = paymentUri.split(":");
+          assert(protocol === "ethereum");
+          const [address, queryString] = rest.split("?");
+          const params = new URLSearchParams(queryString);
+
+          const parsedUri = PaymentUriSchema.parse({
+            protocol,
+            address,
+            amount: params.get("amount") || "",
+          });
+
+          dispatcher.dispatch({
+            name: "bitrefill",
+            address: getAddress(parsedUri.address),
+            amount: parsedUri.amount,
+          });
+        } catch (error) {
+          console.error("[BITREFILL] Failed to parse payment URI:", error);
+        }
+        break;
+      }
+      case "invoice_update":
+      case "invoice_complete": {
+        console.log(`[BITREFILL] ${event} ${paymentUri}`);
+        dispatcher.dispatch({
+          name: "hideBottomSheet",
+        });
+        break;
+      }
+      default:
+        console.log(`[BITREFILL] event ${event}`);
+        break;
+    }
+  };
+
+  const availableBitrefillLanguages = [
```

Comment:
It's used to create the `hl={languageCode}` parameter - bit refill only supports a small set of languages for that, but if you pass a language that is not supported it displays an ugly error, so `hl` is the user's locale language if it's supported by bitrefill, or `en` otherwise.

Reply:
added a comment explaining